prov/efa/src/rdm/efa_rdm_msg.c

```diff
@@ -90,7 +90,8 @@ int efa_rdm_msg_select_rtm(struct efa_rdm_ep *efa_rdm_ep, struct efa_rdm_ope *tx
 
 	if (txe->total_len >= hmem_info[iface].min_read_msg_size &&
 		efa_rdm_interop_rdma_read(efa_rdm_ep, txe->peer) &&
-		(txe->desc[0] || efa_is_cache_available(efa_rdm_ep_domain(efa_rdm_ep))))
+		(txe->desc[0] || efa_is_cache_available(efa_rdm_ep_domain(efa_rdm_ep))) &&
+		use_p2p)
```

Comment:
Does it make more sense to refactor the whole conditional s.t. `use_p2p` is first? That way we can short-circuit and avoid the remaining (potentially expensive) logic.